extendeddatacrossword.go

```diff
@@ -12,34 +13,27 @@ const (
 	column = 1
 )
 
-// ByzantineRowError is thrown when there is a repaired row does not match the expected row merkle root.
-type ByzantineRowError struct {
-	RowNumber      uint
-	LastGoodSquare ExtendedDataSquare
+// ErrUnrepairableDataSquare is thrown when there is insufficient chunks to repair the square.
+var ErrUnrepairableDataSquare = errors.New("failed to solve data square")
+
+// ErrByzantineRow is thrown when there is a repaired row does not match the expected row merkle root.
+type ErrByzantineRow struct {
+	RowNumber uint
 }
 
-func (e *ByzantineRowError) Error() string {
+func (e *ErrByzantineRow) Error() string {
 	return fmt.Sprintf("byzantine row: %d", e.RowNumber)
 }
 
-// ByzantineColumnError is thrown when there is a repaired column does not match the expected column merkle root.
-type ByzantineColumnError struct {
-	ColumnNumber   uint
-	LastGoodSquare ExtendedDataSquare
```

Comment:
Is it really OK to remove this `LastGoodSquare`? It's currently not really used but I'm not sure if this will be needed to proof that this column was actually byzantine? (cc @adlerjohn)

Reply:
I don't really see the need for returning a backup copy?

Reply:
Yeah, that's what I thought too. Thanks for confirming. The data is in the hands of the caller anyways. If required, the caller can construct everything from the row/col index and the data again (not sure if they need to in any case though).

It would help if the spec actually explained under what circumstances nodes and (which nodes?) would construct a BadEncodingProof from what data. Should I open an issue about that?

Reply:
> Should I open an issue about that?

Maybe a comment in https://github.com/lazyledger/lazyledger-specs/issues/23 ? I'm going to start working on Evidence relatively soon.

Reply:
https://github.com/lazyledger/lazyledger-specs/issues/23#issuecomment-812879786
Is that clear enough?

Reply:
Yes, thanks.